File: typescript/examples/basicUsage.ts
/**
 * Basic usage example for DataGrout Conduit
 */

import { Client } from '@datagrout/conduit';

async function main() {
  // Initialize client (uses JSONRPC transport by default)
  const client = new Client({
    url: 'https://gateway.datagrout.ai/servers/YOUR_SERVER_UUID/mcp',
    auth: { bearer: 'YOUR_API_KEY' },
  });

  await client.connect();

  try {
    // List tools (automatically filtered to DataGrout tools)
    console.log('=== Available Tools ===');
    const tools = await client.listTools();
    for (const tool of tools) {
      console.log(`  - ${tool.name}: ${tool.description}`);
    }

    // Call a tool (standard MCP method)
    console.log('\n=== Calling Tool ===');
    const result = await client.callTool('salesforce@1/get_lead@1', {
      id: '00Q5G00000ABC123',
    });
    console.log('Result:', result);

    // Check receipt
    const receipt = client.getLastReceipt();
    if (receipt) {
      console.log(`\nCredits used: ${receipt.actualCredits}`);
      console.log(`Estimated: ${receipt.estimatedCredits}`);
      console.log(`Savings: ${receipt.savings}`);
    }
  } finally {
    await client.disconnect();
  }
}

main().catch(console.error);

File: python/examples/basic_usage.py
"""Basic usage example for DataGrout Conduit."""

import asyncio
from datagrout.conduit import Client


async def main():
    """Demonstrate basic Conduit usage."""
    
    # Initialize client (uses MCP transport by default)
    async with Client(
        "https://gateway.datagrout.ai/servers/YOUR_SERVER_UUID/mcp",
        auth={"bearer": "YOUR_API_KEY"}
    ) as client:
        
        # List tools (automatically filtered to DataGrout tools)
        print("=== Available Tools ===")
        tools = await client.list_tools()
        for tool in tools:
            print(f"  - {tool['name']}: {tool['description']}")
        
        # Call a tool (standard MCP method)
        print("\n=== Calling Tool ===")
        result = await client.call_tool(
            "salesforce@1/get_lead@1",
            {"id": "00Q5G00000ABC123"}
        )
        print(f"Result: {result}")
        
        # Check receipt
        receipt = client.get_last_receipt()
        if receipt:
            print(f"\nCredits used: {receipt.actual_credits}")
            print(f"Estimated: {receipt.estimated_credits}")
            print(f"Savings: {receipt.savings}")


if __name__ == "__main__":
    asyncio.run(main())
